src/main/java/org/logstash/plugins/inputs/http/HttpInitializer.java

```diff
@@ -0,0 +1,42 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.*;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.util.concurrent.DefaultEventExecutorGroup;
+import org.logstash.plugins.inputs.http.util.SslSimpleBuilder;
+
+/**
+ * Created by joaoduarte on 11/10/2017.
+ */
+public class HttpInitializer extends ChannelInitializer<SocketChannel> {
+    private final IMessageHandler messageHandler;
+    private SslSimpleBuilder sslSimpleBuilder;
+    private final DefaultEventExecutorGroup executorGroup;
+
+    public HttpInitializer(IMessageHandler messageHandler, DefaultEventExecutorGroup executorGroup) {
+        this.messageHandler = messageHandler;
+        this.executorGroup = executorGroup;
+    }
+
+    protected void initChannel(SocketChannel socketChannel) throws Exception {
+        ChannelPipeline pipeline = socketChannel.pipeline();
+
+        if(sslSimpleBuilder != null) {
+            SslHandler sslHandler = sslSimpleBuilder.build(socketChannel.alloc());
+            pipeline.addLast(sslHandler);
+        }
+        pipeline.addLast(new HttpServerCodec());
+        pipeline.addLast(new HttpContentDecompressor());
+        pipeline.addLast(new HttpObjectAggregator(512 * 1024));
```

Comment:
Does this mean that if someone is trying to stream us chucked data, we have a max chunk size of 512Kb ? Does this 512Kb limit also apply to non-chunked requests ?

Should we add tests for chunked requests?